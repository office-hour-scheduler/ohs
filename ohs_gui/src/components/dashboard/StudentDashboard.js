import React from 'react';
import shortid from 'shortid';
import PropTypes from 'prop-types';

import LectureSectionCard from './LectureSectionCard';

class StudentDashboard extends React.PureComponent {
  render() {
    const { sections } = this.props;
    return (
      <div id="sections">
        <h1>Current Courses</h1>
        {sections.map(s => (
          <LectureSectionCard section={s} key={shortid.generate()} />
        ))}
      </div>
    );
  }
}

StudentDashboard.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  sections: PropTypes.array.isRequired
};

export default StudentDashboard;
