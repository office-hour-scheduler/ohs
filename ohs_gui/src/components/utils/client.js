import { ApolloClient } from 'apollo-client';
import { createHttpLink } from 'apollo-link-http';
import { setContext } from 'apollo-link-context';
import { InMemoryCache } from 'apollo-cache-inmemory';

const BASE_URL = `${process.env.REACT_APP_INSTRUCTOR_SERVICE_URL ||
  'http://localhost:8000'}`;

const getClient = () => {
  const httpLink = createHttpLink({
    uri: `${BASE_URL}/graphql`
  });

  const authLink = setContext((_, { headers }) => {
    // get the authentication token from local storage if it exists
    const token = window.sessionStorage.getItem('token');
    // return the headers to the context so httpLink can read them
    return {
      headers: {
        ...headers,
        Authorization: token ? `Bearer ${token}` : ''
      }
    };
  });

  const client = new ApolloClient({
    link: authLink.concat(httpLink),
    cache: new InMemoryCache()
  });

  return client;
};

const client = null;

export { client, BASE_URL, getClient };
