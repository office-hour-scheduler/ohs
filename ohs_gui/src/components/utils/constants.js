// User roles
const PROFESSOR = 1;
const STUDENT = 2;

export { PROFESSOR, STUDENT };
