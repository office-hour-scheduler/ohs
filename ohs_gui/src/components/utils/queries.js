import gql from 'graphql-tag';

const GET_COURSES = gql`
  query getCourses {
    courses {
      courseCode
    }
  }
`;

  const GET_MEETINGS = gql`
  query getMeeting($meetingId: String!) {
    meeting(meetingId: $meetingId) {
    meeting_id
    office_hour_id
    index
    instructor
    student
    notes
    comments
    start_time
    }
  }
`;

const GET_UPCOMING_MEETINGS = gql`
  query getUpcomingMeetings {
    upcomingMeetings {
      meetingId
      officeHourId
      index
      instructor {
        firstName
        lastName
        userName
      }
      student {
        firstName
        lastName
        studentNumber
      }
      startTime
    }
  }
`;

const GET_SECTIONS = gql`
  query getSections {
    sections {
      course {
        courseCode
      }
      year
      semester
      sectionCode
    }
  }
`;

const GET_SECTION = gql`
  query getSection(
    $course: CourseInput!
    $year: Int!
    $semester: Semester!
    $sectionCode: String!
  ) {
    section(
      course: $course
      year: $year
      semester: $semester
      sectionCode: $sectionCode
    ) {
      course {
        courseCode
      }
      year
      semester
      sectionCode
      numStudents
    }
  }
`;

const GET_SECTIONS_FOR_STUDENT = gql`
  query getSectionForStudent($studentNum: String!) {
    sections(enrolledIn: $studentNum) {
      course {
        courseCode
      }
      year
      semester
      sectionCode
      numStudents
    }
  }
`;

const GET_SECTIONS_FOR_COURSE = gql`
  query getSectionForCourse($courseCode: String!, $taughtBy: String!) {
    sections(courseCode: $courseCode, taughtBy: $taughtBy) {
      course {
        courseCode
      }
      year
      semester
      sectionCode
      numStudents
    }
  }
`;

export {
  GET_COURSES,
  GET_SECTIONS,
  GET_SECTION,
  GET_SECTIONS_FOR_COURSE,
  GET_UPCOMING_MEETINGS,
  GET_SECTIONS_FOR_STUDENT,
  GET_MEETINGS
};
