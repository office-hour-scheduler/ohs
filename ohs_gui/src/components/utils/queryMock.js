// @flow
import { QueryMock } from 'graphql-query-test-mock';

const queryMock = new QueryMock();

export default queryMock;
