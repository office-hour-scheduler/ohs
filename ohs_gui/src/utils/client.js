import ApolloClient from "apollo-boost";

const client = new ApolloClient({
    uri: "http://127.0.0.1:8000/graphql"
});

export {client};