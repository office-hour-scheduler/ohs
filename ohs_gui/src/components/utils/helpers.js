import { roles } from './constants';

function userIsProf(user) {
  return user && user.role === roles.PROFESSOR;
}

function getSemesterCode(sem) {
  // assumption: summer courses are assumed to be full session
  switch (sem) {
    case 'SUMMER':
    case 'FULL_YEAR':
      return 'Y';
    case 'WINTER':
      return 'S';
    case 'FALL':
      return 'F';
    default:
      return '';
  }
}

function getFormattedSectionName(section) {
  return `${section.course.courseCode}H1${getSemesterCode(
    section.semester
  )} - ${section.sectionCode} - ${section.year}`;
}

function numberToDayOfWeek(dayNumber) {
  switch(dayNumber) {
    case 0:
      return "SUNDAY";
    case 1:
      return "MONDAY";
    case 2:
      return "TUESDAY";
    case 3:
      return "WEDNESDAY";
    case 4:
      return "THURSDAY";
    case 5:
      return "FRIDAY";
    case 6:
      return "SATRUDAY";
    default:
      return '';
  }
}

export { userIsProf, getSemesterCode, getFormattedSectionName, numberToDayOfWeek };
