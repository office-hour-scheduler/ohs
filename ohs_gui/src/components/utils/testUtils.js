import { roles } from './constants';

const PROF_USER = {
  firstName: 'Albert',
  lastName: 'Einstein',
  role: roles.PROFESSOR,
  userName: 'emc2'
};

const STUDENT_USER = {
  firstName: 'Bernie',
  lastName: 'Sanders',
  role: roles.STUDENT,
  userName: 'sensan'
};

const SECTIONS = [
  {
    course: {
      courseCode: 'CSC302'
    },
    year: 2019,
    semester: 'FALL',
    sectionCode: 'LEC0101',
    numStudents: 25
  },
  {
    course: {
      courseCode: 'CSC304'
    },
    year: 2019,
    semester: 'FALL',
    sectionCode: 'LEC0102',
    numStudents: 25
  }
];

export { SECTIONS, PROF_USER, STUDENT_USER };
