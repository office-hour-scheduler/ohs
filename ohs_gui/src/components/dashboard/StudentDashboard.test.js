import React from 'react';
import { shallow } from 'enzyme';

import StudentDashboard from './StudentDashboard';
import { SECTIONS } from '../utils/testUtils';

// need this to deal with randomly generated ids
jest.mock('shortid', () => {
  return {
    generate: jest.fn(() => 1)
  };
});

it('should render correctly with no props', () => {
  const t = () => {
    shallow(<StudentDashboard />);
  };
  expect(t).toThrow(Error);
});

it('test if it renders with params', () => {
  const component = shallow(<StudentDashboard sections={SECTIONS} />);
  expect(component).toMatchSnapshot();
});

it('test if it renders with empty list', () => {
  const component = shallow(<StudentDashboard sections={[]} />);
  expect(component).toMatchSnapshot();
});
