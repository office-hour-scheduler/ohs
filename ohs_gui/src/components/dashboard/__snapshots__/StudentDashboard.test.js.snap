// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if it renders with empty list 1`] = `
<div
  id="sections"
>
  <h1>
    Current Courses
  </h1>
</div>
`;

exports[`test if it renders with params 1`] = `
<div
  id="sections"
>
  <h1>
    Current Courses
  </h1>
  <LectureSectionCard
    key="1"
    section={
      Object {
        "course": Object {
          "courseCode": "CSC302",
        },
        "numStudents": 25,
        "sectionCode": "LEC0101",
        "semester": "FALL",
        "year": 2019,
      }
    }
  />
  <LectureSectionCard
    key="1"
    section={
      Object {
        "course": Object {
          "courseCode": "CSC304",
        },
        "numStudents": 25,
        "sectionCode": "LEC0102",
        "semester": "FALL",
        "year": 2019,
      }
    }
  />
</div>
`;
